packages/react-next/src/components/OverflowSet/OverflowSet.base.tsx

```diff
@@ -1,257 +1,225 @@
+            disabled: keytip.disabled || !!(overflowItem.disabled || overflowItem.isDisabled),
+            hasDynamicChildren: keytip.hasDynamicChildren,
+            hasMenu: keytip.hasMenu,
+          };
+          if (keytip.hasDynamicChildren || getSubMenuForItem(overflowItem)) {
+            // If the keytip has a submenu or children nodes, change onExecute to persistedKeytipExecute
+            persistedKeytip.onExecute = keytipManager.menuExecute.bind(
+              keytipManager,
+              overflowKeytipSequences,
+              overflowItem.keytipProps.keySequences,
+            );
           } else {
-            // Nothing to change, add overflowItem to list
-            newOverflowItems.push(overflowItem);
+            // If the keytip doesn't have a submenu, just execute the original function
+            persistedKeytip.onExecute = keytip.onExecute;
           }
-        });
-    } else if (items) {
-      newOverflowItems = items;
+          // Add this persisted keytip to our internal list, use a temporary uniqueID (its content)
+          // uniqueID will get updated on register
+          persistedKeytips[persistedKeytip.content] = persistedKeytip;
+          // Add the overflow sequence to this item
+          const newOverflowItem = {
+            ...overflowItem,
+            keytipProps: {
+              ...keytip,
+              overflowSetSequence: overflowKeytipSequences,
+            },
+          };
+          newOverflowItems.push(newOverflowItem);
+        } else {
+          // Nothing to change, add overflowItem to list
+          newOverflowItems.push(overflowItem);
+        }
+      });
+    } else {
+      newOverflowItems = itemsProp;
     }
-    return <div {...wrapperDivProps}>{this.props.onRenderOverflowButton(newOverflowItems)}</div>;
+    // Set up the keytip effect here, after persistedKeytips
+    useKeytipRegistrations(persistedKeytips, keytipManager);
+    return <div {...wrapperDivProps}>{props.onRenderOverflowButton(newOverflowItems)}</div>;
   };
 
-  /**
-   * Gets the subMenu for an overflow item
-   * Checks if itemSubMenuProvider has been defined, if not defaults to subMenuProps
-   */
+  // Gets the subMenu for an overflow item
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  private _getSubMenuForItem(item: IOverflowSetItemProps): any[] | undefined {
-    if (this.props.itemSubMenuProvider) {
-      return this.props.itemSubMenuProvider(item);
+  const getSubMenuForItem = (item: any): any[] | undefined => {
+    // Checks if itemSubMenuProvider has been defined, if not defaults to subMenuProps
+    if (props.itemSubMenuProvider) {
+      return props.itemSubMenuProvider(item);
     }
     if (item.subMenuProps) {
       return item.subMenuProps.items;
     }
     return undefined;
+  };
+
+  useConst(() => {
+    warnMutuallyExclusive(COMPONENT_NAME, props, {
+      doNotContainWithinFocusZone: 'focusZoneProps',
+    });
+  });
```

Comment:
Do `useWarnings` instead (wrapped in `if (process.env.NODE_ENV !== 'production')`)